### DISCUSSION TITLE
output the Extreme Value of material defined on surface

### DISCUSSION THREAD

**Discussion by echo1115** [idaholab/moose#24162] [Q&A General] [answered]
Hi all,
I want to know how to get the the Extreme Value of material defined on surface by using postprocessor.

```
[AuxVariables]
  [./oxide_thickness]
    #order = CONSTANT
    #family = MONOMIAL
  [../]
[]

[AuxKernels]
  [./oxide]
    type = MaterialRealAux
    property = oxide_thickness
    variable = oxide_thickness
    execute_on = 'initial linear'
  [../]
[]

[Materials]
  [./oxide_thickness]
    type = OxidationMaterial
    boundary = side1
  [../]
[]

[Postprocessors]
  [./max_oxide_thickness1]
    type = ElementExtremeMaterialProperty
    mat_prop =  oxide_thickness
    execute_on = 'initial timestep_end'
  [../]
  [./max_oxide_thickness2]
    type = NodalExtremeValue
    variable = oxide_thickness
    execute_on = 'initial timestep_end'
  [../]
[]
```
the material `oxide_thickness ` is defined on boundary `side1` (the side1 is part of `block1 `) and I want to obtain its maximum value. 

When using the `ElementExtremeMaterialProperty` PP, it said that

>  `oxide_thickness ` was not defined on block1

While using `NodalExtremeValue` and setting the `AuxVariables `to First order Lagrangian, it shows:

> Nodal AuxKernel 'oxide_thickness' attempted to reference material property 'property'
> Consider using an elemental auxiliary variable for 'oxide_thickness'

**Comment by GiudGiud** [comment]
Hello

There's no nodal material property, the "boundary" for the material property really refers to quadrature points on element sides on the boundary.

Can you try using a constant monomial and seeing if the code naturally only populates the DOFs near the side?

Otherwise there's several things we could try to add here, but will need some developing
- SideExtremeMaterialProperty PP
- some nodal projection to MaterialRealAux

Guillaume

**Reply by echo1115** [reply]
the `oxide_thickness` is in the elements near the side:
![oxide](https://user-images.githubusercontent.com/69078335/234011155-2c3cb6e5-2104-48e9-bf0c-9a31c0be4e14.JPG)

the problem solved when I change into the following input:
```
[Materials]
  [./oxide_thickness]
    type = OxidationMaterial
    output_properties = 'oxide_thickness'   #added
    outputs = all     #added
    boundary = side1
  [../]
[]

[Postprocessors]
  [./max_oxide]
    type = ElementExtremeValue
    variable = oxide_thickness
    execute_on = 'initial timestep_end'
  [../]
[]
```
And the related `[AuxVariables]` and  `[AuxKernels]` were deleted.

Thanks for your help. @GiudGiud

**Reply by GiudGiud** [reply, marked as answer]
sounds good. This is equivalent to MaterialRealAux + const monomial auxvariable, it just happens in the background due to matproperty output

**Reply by echo1115** [reply]
Can I ask another question？
Except `none ` and `all`, what else can the `outputs `parameter be set in the `Materials` ?

**Reply by GiudGiud** [reply]
any name of Outputs object in the [Outputs] block. For example, below `final_exodus`
```
[Outputs]
  [final_exodus]
     type = Exodus
     execute_on = FINAL
  []
[]
```

**Reply by echo1115** [reply]
Thanks for your reply.

**Reply by echo1115** [reply]
When the material object is included in Action, it seems the parameter `output_properties` and  `outputs` don't work. Will a `SideExtremeMaterialProperty ` PP be developed in MOOSE in the future?

**Reply by GiudGiud** [reply]
You can still use the auxvariable/auxkernel route though. 

We have no immediate plans on this object afaik. We d welcome the contribution though
